test/integration/test-video-players-helper.js

```diff
@@ -0,0 +1,177 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {listenOncePromise} from '../../src/event-helper';
+import {timerFor} from '../../src/timer';
+import {viewportForDoc} from '../../src/viewport';
+import {VideoInterface, VideoEvents} from '../../src/video-interface';
+import {supportsAutoplay} from '../../src/service/video-manager-impl';
+import {
+  createFixtureIframe,
+  expectBodyToBecomeVisible,
+  poll,
+} from '../../testing/iframe';
+
+export function runVideoPlayerIntegrationTests(createVideoElementFunc) {
+
+  it('should override the video interface methods', function() {
+    return getVideoPlayer(/* opt_outsideView */ false,
+        /* opt_dontWaitForLoad */ true)
+    .then(v => {
+      const impl = v.implementation_;
+      const methods = Object.getOwnPropertyNames(
```

Comment:
as per @lannka talk today, this seems like a lot of logic for a test. Might be simpler to just explicitly expect methods to exist here by name instead of iterating over the implementation.